sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2269,6 +2270,8 @@ public <T> Mono<FeedResponse<T>> readMany(
                                 List<T> finalList = new ArrayList<>();
                                 HashMap<String, String> headers = new HashMap<>();
                                 ConcurrentMap<String, QueryMetrics> aggregatedQueryMetrics = new ConcurrentHashMap<>();
+                                List<ClientSideRequestStatistics> aggregateRequestStatistics = new ArrayList<>();
```

Comment:
We should synchronize this list. Common way to do this would be: 
```suggestion
                                List<ClientSideRequestStatistics> aggregateRequestStatistics = Collections.synchronizedList(new ArrayList<>());
```

Reply:
Happy to make the change, but would love to know why.  This isn't modified in a multi-threaded fashion below, and they end up getting added to the already synchronized collection in the `FeedResponseDiagnostics` [here](https://github.com/Azure/azure-sdk-for-java/blob/99ab4a0b6dcf5062899ce90899cb29084ae21f65/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java#L130), so I'm not sure what gain we would get out of it.

Reply:
Yeah, valid point. The only reason I thought we should make it synchronized is because the `aggregatedQueryMetrics` map is also concurrent hash map. But it looks like we are only adding new values to this in a sequential manner.

Reply:
aggregatedQueryMetrics is concurrent because the constructor requires it to be so.  Going to leave this as is.